// steelservice-domain/generated/prisma-client/index-browser.js

Object.defineProperty(exports, "__esModule", { value: true });

const {
  Decimal,
  objectEnumValues,
  makeStrictEnum,
  Public,
  getRuntime,
  skip
} = require('./runtime/index-browser.js')


const Prisma = {}

exports.Prisma = Prisma
exports.$Enums = {}

/**
 * Prisma Client JS version: 5.22.0
 * Query Engine version: 605197351a3c8bdd595af2d2a9bc3025bca48ea2
 */
Prisma.prismaVersion = {
  client: "5.22.0",
  engine: "605197351a3c8bdd595af2d2a9bc3025bca48ea2"
}

Prisma.PrismaClientKnownRequestError = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`PrismaClientKnownRequestError is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)};
Prisma.PrismaClientUnknownRequestError = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`PrismaClientUnknownRequestError is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.PrismaClientRustPanicError = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`PrismaClientRustPanicError is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.PrismaClientInitializationError = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`PrismaClientInitializationError is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.PrismaClientValidationError = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`PrismaClientValidationError is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.NotFoundError = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`NotFoundError is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.Decimal = Decimal

/**
 * Re-export of sql-template-tag
 */
Prisma.sql = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`sqltag is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.empty = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`empty is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.join = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`join is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.raw = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`raw is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.validator = Public.validator

/**
* Extensions
*/
Prisma.getExtensionContext = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`Extensions.getExtensionContext is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.defineExtension = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`Extensions.defineExtension is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}

/**
 * Shorthand utilities for JSON filtering
 */
Prisma.DbNull = objectEnumValues.instances.DbNull
Prisma.JsonNull = objectEnumValues.instances.JsonNull
Prisma.AnyNull = objectEnumValues.instances.AnyNull

Prisma.NullTypes = {
  DbNull: objectEnumValues.classes.DbNull,
  JsonNull: objectEnumValues.classes.JsonNull,
  AnyNull: objectEnumValues.classes.AnyNull
}



/**
 * Enums
 */

exports.Prisma.TransactionIsolationLevel = makeStrictEnum({
  ReadUncommitted: 'ReadUncommitted',
  ReadCommitted: 'ReadCommitted',
  RepeatableRead: 'RepeatableRead',
  Serializable: 'Serializable'
});

exports.Prisma.AssetScalarFieldEnum = {
  id: 'id',
  type: 'type',
  assetUrl: 'assetUrl',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt',
  createdById: 'createdById',
  updatedById: 'updatedById',
  deletedById: 'deletedById'
};

exports.Prisma.CalculatorFormulaScalarFieldEnum = {
  id: 'id',
  name: 'name',
  description: 'description',
  calculates: 'calculates',
  formula: 'formula',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt',
  createdById: 'createdById',
  updatedById: 'updatedById',
  deletedById: 'deletedById'
};

exports.Prisma.CountryScalarFieldEnum = {
  id: 'id',
  name: 'name',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt',
  createdById: 'createdById',
  updatedById: 'updatedById',
  deletedById: 'deletedById'
};

exports.Prisma.CustomerScalarFieldEnum = {
  id: 'id',
  description: 'description',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt',
  createdById: 'createdById',
  updatedById: 'updatedById',
  deletedById: 'deletedById'
};

exports.Prisma.EntityTypeScalarFieldEnum = {
  id: 'id',
  name: 'name',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt',
  createdById: 'createdById',
  updatedById: 'updatedById',
  deletedById: 'deletedById'
};

exports.Prisma.MaterialScalarFieldEnum = {
  id: 'id',
  name: 'name',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt',
  createdById: 'createdById',
  updatedById: 'updatedById',
  deletedById: 'deletedById'
};

exports.Prisma.MeasurementUnitScalarFieldEnum = {
  id: 'id',
  unit: 'unit',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt',
  createdById: 'createdById',
  updatedById: 'updatedById',
  deletedById: 'deletedById'
};

exports.Prisma.ProductScalarFieldEnum = {
  id: 'id',
  description: 'description',
  formulaId: 'formulaId',
  materialId: 'materialId',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt',
  createdById: 'createdById',
  updatedById: 'updatedById',
  deletedById: 'deletedById'
};

exports.Prisma.ProfileScalarFieldEnum = {
  userId: 'userId',
  firstName: 'firstName',
  lastName: 'lastName',
  email: 'email',
  phoneNumber: 'phoneNumber',
  addressOne: 'addressOne',
  addressTwo: 'addressTwo',
  postalCode: 'postalCode',
  entityTypeId: 'entityTypeId',
  countryId: 'countryId',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt',
  createdById: 'createdById',
  updatedById: 'updatedById',
  deletedById: 'deletedById'
};

exports.Prisma.QuoteScalarFieldEnum = {
  id: 'id',
  clientId: 'clientId',
  agentId: 'agentId',
  typeId: 'typeId',
  invoiceId: 'invoiceId',
  quoteeInputId: 'quoteeInputId',
  forCustomerId: 'forCustomerId',
  status: 'status',
  chatRoomId: 'chatRoomId',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt',
  createdById: 'createdById',
  updatedById: 'updatedById',
  deletedById: 'deletedById'
};

exports.Prisma.QuoteTypeScalarFieldEnum = {
  id: 'id',
  name: 'name',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt',
  createdById: 'createdById',
  updatedById: 'updatedById',
  deletedById: 'deletedById'
};

exports.Prisma.QuoteeInputScalarFieldEnum = {
  id: 'id',
  comment: 'comment',
  quoteeInternalReference: 'quoteeInternalReference',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt',
  createdById: 'createdById',
  updatedById: 'updatedById',
  deletedById: 'deletedById'
};

exports.Prisma.RegionScalarFieldEnum = {
  id: 'id',
  name: 'name',
  countryId: 'countryId',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt',
  createdById: 'createdById',
  updatedById: 'updatedById',
  deletedById: 'deletedById'
};

exports.Prisma.RequestMessageScalarFieldEnum = {
  id: 'id',
  message: 'message',
  satisfied: 'satisfied',
  quoteId: 'quoteId',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt',
  createdById: 'createdById',
  updatedById: 'updatedById',
  deletedById: 'deletedById'
};

exports.Prisma.RoleScalarFieldEnum = {
  id: 'id',
  name: 'name',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt',
  createdById: 'createdById',
  updatedById: 'updatedById',
  deletedById: 'deletedById'
};

exports.Prisma.ServiceScalarFieldEnum = {
  id: 'id',
  name: 'name',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt',
  createdById: 'createdById',
  updatedById: 'updatedById',
  deletedById: 'deletedById'
};

exports.Prisma.TechnicalAttributeScalarFieldEnum = {
  id: 'id',
  name: 'name',
  var: 'var',
  unitId: 'unitId',
  selectionValues: 'selectionValues',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt',
  createdById: 'createdById',
  updatedById: 'updatedById',
  deletedById: 'deletedById'
};

exports.Prisma.UserScalarFieldEnum = {
  id: 'id',
  banned: 'banned',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.UserActionScalarFieldEnum = {
  id: 'id',
  actionType: 'actionType',
  userId: 'userId',
  targetType: 'targetType',
  targetId: 'targetId',
  timestamp: 'timestamp'
};

exports.Prisma.UserRoleScalarFieldEnum = {
  id: 'id',
  userId: 'userId',
  roleId: 'roleId',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt',
  createdById: 'createdById',
  updatedById: 'updatedById',
  deletedById: 'deletedById'
};

exports.Prisma.ProductAssetScalarFieldEnum = {
  id: 'id',
  productId: 'productId',
  assetId: 'assetId',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.ProductQuoteScalarFieldEnum = {
  id: 'id',
  productId: 'productId',
  quoteId: 'quoteId',
  quantity: 'quantity',
  comment: 'comment',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt',
  createdById: 'createdById',
  updatedById: 'updatedById',
  deletedById: 'deletedById'
};

exports.Prisma.ProductServiceScalarFieldEnum = {
  id: 'id',
  productId: 'productId',
  serviceId: 'serviceId',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt',
  createdById: 'createdById',
  updatedById: 'updatedById',
  deletedById: 'deletedById'
};

exports.Prisma.QuoteAssetScalarFieldEnum = {
  id: 'id',
  quoteId: 'quoteId',
  technicalDocumentsQuoteId: 'technicalDocumentsQuoteId',
  assetId: 'assetId',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt',
  createdById: 'createdById',
  updatedById: 'updatedById',
  deletedById: 'deletedById'
};

exports.Prisma.QuoteProductAttributeQuoteeScalarFieldEnum = {
  id: 'id',
  quoteId: 'quoteId',
  productId: 'productId',
  quoteeProductInput: 'quoteeProductInput',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt',
  createdById: 'createdById',
  updatedById: 'updatedById',
  deletedById: 'deletedById'
};

exports.Prisma.QuoteeProductServiceScalarFieldEnum = {
  id: 'id',
  productId: 'productId',
  quoteId: 'quoteId',
  serviceId: 'serviceId',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt',
  createdById: 'createdById',
  updatedById: 'updatedById',
  deletedById: 'deletedById'
};

exports.Prisma.TechnicalAttributeMaterialScalarFieldEnum = {
  id: 'id',
  materialId: 'materialId',
  attributeId: 'attributeId',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt',
  createdById: 'createdById',
  updatedById: 'updatedById',
  deletedById: 'deletedById'
};

exports.Prisma.TechnicalAttributeProductScalarFieldEnum = {
  id: 'id',
  insertableForQuotation: 'insertableForQuotation',
  productId: 'productId',
  attributeId: 'attributeId',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt',
  createdById: 'createdById',
  updatedById: 'updatedById',
  deletedById: 'deletedById'
};

exports.Prisma.UserCustomerScalarFieldEnum = {
  id: 'id',
  userId: 'userId',
  customerId: 'customerId',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt',
  createdById: 'createdById',
  updatedById: 'updatedById',
  deletedById: 'deletedById'
};

exports.Prisma.SortOrder = {
  asc: 'asc',
  desc: 'desc'
};

exports.Prisma.NullableJsonNullValueInput = {
  DbNull: Prisma.DbNull,
  JsonNull: Prisma.JsonNull
};

exports.Prisma.QueryMode = {
  default: 'default',
  insensitive: 'insensitive'
};

exports.Prisma.NullsOrder = {
  first: 'first',
  last: 'last'
};

exports.Prisma.JsonNullValueFilter = {
  DbNull: Prisma.DbNull,
  JsonNull: Prisma.JsonNull,
  AnyNull: Prisma.AnyNull
};
exports.AssetType = exports.$Enums.AssetType = {
  product_image: 'product_image',
  quote_documentation: 'quote_documentation',
  invoice: 'invoice'
};

exports.QuoteStatus = exports.$Enums.QuoteStatus = {
  draft: 'draft',
  analysis: 'analysis',
  offer: 'offer',
  accepted: 'accepted',
  pending: 'pending',
  delivered: 'delivered'
};

exports.ActionType = exports.$Enums.ActionType = {
  Created: 'Created',
  Updated: 'Updated',
  Deleted: 'Deleted'
};

exports.Prisma.ModelName = {
  Asset: 'Asset',
  CalculatorFormula: 'CalculatorFormula',
  Country: 'Country',
  Customer: 'Customer',
  EntityType: 'EntityType',
  Material: 'Material',
  MeasurementUnit: 'MeasurementUnit',
  Product: 'Product',
  Profile: 'Profile',
  Quote: 'Quote',
  QuoteType: 'QuoteType',
  QuoteeInput: 'QuoteeInput',
  Region: 'Region',
  RequestMessage: 'RequestMessage',
  Role: 'Role',
  Service: 'Service',
  TechnicalAttribute: 'TechnicalAttribute',
  User: 'User',
  UserAction: 'UserAction',
  UserRole: 'UserRole',
  ProductAsset: 'ProductAsset',
  ProductQuote: 'ProductQuote',
  ProductService: 'ProductService',
  QuoteAsset: 'QuoteAsset',
  QuoteProductAttributeQuotee: 'QuoteProductAttributeQuotee',
  QuoteeProductService: 'QuoteeProductService',
  TechnicalAttributeMaterial: 'TechnicalAttributeMaterial',
  TechnicalAttributeProduct: 'TechnicalAttributeProduct',
  UserCustomer: 'UserCustomer'
};

/**
 * This is a stub Prisma Client that will error at runtime if called.
 */
class PrismaClient {
  constructor() {
    return new Proxy(this, {
      get(target, prop) {
        let message
        const runtime = getRuntime()
        if (runtime.isEdge) {
          message = `PrismaClient is not configured to run in ${runtime.prettyName}. In order to run Prisma Client on edge runtime, either:
- Use Prisma Accelerate: https://pris.ly/d/accelerate
- Use Driver Adapters: https://pris.ly/d/driver-adapters
`;
        } else {
          message = 'PrismaClient is unable to run in this browser environment, or has been bundled for the browser (running in `' + runtime.prettyName + '`).'
        }
        
        message += `
If this is unexpected, please open an issue: https://pris.ly/prisma-prisma-bug-report`

        throw new Error(message)
      }
    })
  }
}

exports.PrismaClient = PrismaClient

Object.assign(exports, Prisma)
